rust/stub/src/fat.rs

```diff
@@ -0,0 +1,69 @@
+use alloc::vec::Vec;
+use uefi::{prelude::*, CString16, Result};
+
+use crate::common::{boot_linux_unchecked, extract_string};
+use crate::pe_section::pe_section;
+use crate::uefi_helpers::booted_image_file;
+
+/// Extract bytes from a PE section.
+pub fn extract_bytes(pe_data: &[u8], section: &str) -> Result<Vec<u8>> {
+    let bytes: Vec<u8> = pe_section(pe_data, section)
+        .ok_or(Status::INVALID_PARAMETER)?
+        .try_into()
+        .map_err(|_| Status::INVALID_PARAMETER)?;
+
+    Ok(bytes)
+}
+
+/// The configuration that is embedded at build time.
+///
+/// After this stub is built, lzbt needs to embed configuration into the binary by adding PE
+/// sections. This struct represents that information.
+struct EmbeddedConfiguration {
+    /// The kernel command-line.
+    cmdline: CString16,
+
+    /// The kernel as raw bytes.
+    kernel: Vec<u8>,
+
+    /// The initrd as raw bytes.
+    initrd: Vec<u8>,
+}
+
+impl EmbeddedConfiguration {
+    fn new(file_data: &[u8]) -> Result<Self> {
+        Ok(Self {
+            kernel: extract_bytes(file_data, ".linux")?,
+            initrd: extract_bytes(file_data, ".initrd")?,
+            cmdline: extract_string(file_data, ".cmdline")?,
+        })
+    }
+}
+
+pub fn boot_linux(handle: Handle, mut system_table: SystemTable<Boot>) -> Status {
+    uefi_services::init(&mut system_table).unwrap();
+
+    log::info!("Booting a fat UKI.");
```

Comment:
I'm not sure those log are useful, aren't they breaking the quiet thing too?

Reply:
(maybe keep them actually, because we have max level warn on prod anyway)

Reply:
Removed them. They were also kind of in the wrong spot.